Merge sfrinaldi/update-structure into main

sfrinaldi opened the pull request:
Adjusting structure to match the adjustments made in [config_stp](https://github.com/uasal/config_stp/tree/develop) for packaging.

sfrinaldi commented:
Do we still want to have the statement that tools in the config directory should be in separate sub-directories? Is this still applicable with how the repositories are being organized? _(Doesn't seem to be the case for how the current config repositories are looking presently.)_

AdamSchilperoort commented:
Ah, I removed extra subdirectories in the other repos. I thought it was a bit redundant to have a name of a directory that matches the toml file name, which also happens to match a [section_name], so instead opted for a flat structure. We could keep a directory structure if there's a good reason to separate files into specific categories. It doesn't really matter to the parser either way since it'll load everything into a single dictionary

sfrinaldi commented:
[like]  Rinaldi, Stephanie - (srinaldi) reacted to your message:
________________________________
From: Adam Schilperoort ***@***.***>
Sent: Friday, February 21, 2025 6:25:55 AM
To: uasal/config_project_template ***@***.***>
Cc: Rinaldi, Stephanie - (srinaldi) ***@***.***>; Assign ***@***.***>
Subject: [EXT] Re: [uasal/config_project_template] Merge sfrinaldi/update-structure into main (PR #5)


Ah, I removed extra subdirectories in the other repos. I thought it was a bit redundant to have a name of a directory that matches the toml file name, which also happens to match a [section_name], so instead opted for a flat structure. We could keep a directory structure if there's a good reason to separate files into specific categories. It doesn't really matter to the parser either way since it'll load everything into a single dictionary

—
Reply to this email directly, view it on GitHub<https://url.usb.m.mimecastprotect.com/s/StRuCZZEkxi7EkvV9Czf0TBd9ZZ?domain=github.com>, or unsubscribe<https://url.usb.m.mimecastprotect.com/s/x1fkC1VkgOf61PgDNSGhyTVNnll?domain=github.com>

External Email

________________________________
.
You are receiving this because you were assigned.Message ID: ***@***.***>

[AdamSchilperoort]AdamSchilperoort left a comment (uasal/config_project_template#5)<https://url.usb.m.mimecastprotect.com/s/StRuCZZEkxi7EkvV9Czf0TBd9ZZ?domain=github.com>

Ah, I removed extra subdirectories in the other repos. I thought it was a bit redundant to have a name of a directory that matches the toml file name, which also happens to match a [section_name], so instead opted for a flat structure. We could keep a directory structure if there's a good reason to separate files into specific categories. It doesn't really matter to the parser either way since it'll load everything into a single dictionary

—
Reply to this email directly, view it on GitHub<https://url.usb.m.mimecastprotect.com/s/StRuCZZEkxi7EkvV9Czf0TBd9ZZ?domain=github.com>, or unsubscribe<https://url.usb.m.mimecastprotect.com/s/x1fkC1VkgOf61PgDNSGhyTVNnll?domain=github.com>.
You are receiving this because you were assigned.Message ID: ***@***.***>

patrickingraham commented on README.md:
I think the title needs to change to be `config_project_template`

sfrinaldi replied:
Should be applied now / changed in other locations too for consistency

patrickingraham replied:
ah, but i meant in line 1 of this file

sfrinaldi replied:
Ah! Thanks / read the title quickly apparently and just saw this highlighting the other line.

patrickingraham commented on README.md:
```diff
@@ -5,13 +5,15 @@ The naming convention should be config_project_toolname. So a hypothetical examp
+- Values without a unit will be marked as *unitless*.
 - The origin of the values should be in a comment (for now).
 - At this time, no other filenames should start with an `_`. This functionality is reserved for future feature implementation(s).
 - Any added configuration file must contain the *full set* of available parameters and not just overrides for the defaults.
```
Just a heads up that I moved the FAQ into the description of configuration management which is currently here:
https://github.com/uasal/lab_documents/blob/pingraham/config_description/computing/development_guide/configuration_management.md

We should link that description from here and remove the FAQ, but I think the above needs to be merged first

sfrinaldi replied:
Applied a todo item for that then. Did remove the tool subdirectory part per today's sprint conversation as a heads up. Changed config_project to config_project_template.

sfrinaldi replied:
Did this change actually since I noticed the configuration summary went through on lab_documents. Should be good now!

sfrinaldi commented:
@patrickingraham Template should be like the structure for the other config repos now I believe. (Such as: https://github.com/uasal/config_stp_wcc/tree/develop, https://github.com/uasal/config_stp_wcc/tree/develop, and https://github.com/uasal/config_stp_esc.) Though looks like some of those repos either don't have a docs and or a notebooks directory (or not merged in yet). This is separate from this PR, but do we want to add placeholders to those config repos for those directories?

patrickingraham commented on config_project_template/config_loader.py:
should this not be removed, and presumably the __init__.py ?

sfrinaldi replied:
Right now this is duplicated (the config_loader.py) for the config repos. Think we would want this here as well as the init.py for reference. I can remove though if otherwise!

patrickingraham commented on tests/test_config_files.py:
I don't think this is correct. 
I think you want the tests from @AdamSchilperoort 's new PR.
https://github.com/uasal/config_stp/pull/26/files#diff-96a8c92ae08ea3e6b29838daadc7f297741a942a1f8980aafed4d6920e17c864

sfrinaldi replied:
Yes! I didn't see it yet at the time of adjusting this.

sfrinaldi replied:
Updated / added comments for what items would need to be changed for reference (tool name parts).

sfrinaldi commented on README.md:
```diff
@@ -5,13 +5,15 @@ The naming convention should be config_project_toolname. So a hypothetical examp
-- configuration parameters that are common to all tools in the repository belong in the common.toml file. 
-- There should be one directory per tool (e.g. tool1, tool2). 
+- Configuration parameters that are common to all tools in the repository belong in the `common.toml` file. 
+- There should be one directory per tool (e.g. tool1, tool2) within the `configs` directory. 
```
Ah! Thought I removed all references to it. Thanks!
